lib/Twig/Environment.php

```diff
@@ -326,15 +363,21 @@ public function loadTemplate($name, $index = null)
         }
 
         if (!class_exists($cls, false)) {
-            if (false === $cache = $this->getCacheFilename($name)) {
-                eval('?>'.$this->compileSource($this->getLoader()->getSource($name), $name));
+            if ($this->bcGetCacheFilename) {
+                $key = $this->getCacheFilename($name);
             } else {
-                if (!is_file($cache) || ($this->isAutoReload() && !$this->isTemplateFresh($name, filemtime($cache)))) {
-                    $this->writeCacheFile($cache, $this->compileSource($this->getLoader()->getSource($name), $name));
-                }
+                $key = $this->cache->generateKey($cls, $this->templateClassPrefix);
```

Comment:
This change from `getCacheFilename($name)` to `$this->cache->generateKey($cls, $this->templateClassPrefix)` is proving to be a bit of a problem in the end goal of trying *not* to override loadTemplate() in Drupal 8. We use `$name` as part of the cache key.